src/app/snapp_test_transaction/snapp_test_transaction.ml

```diff
@@ -0,0 +1,1247 @@
+               sprintf "\"%s\"" (Snark_params.Tick.Field.to_string s)))
+  in
+  let verification_key (update : Party.Update.t) =
+    match update.verification_key with
+    | Snapp_basic.Set_or_keep.Keep ->
+        "null"
+    | Set vk_with_hash ->
+        sprintf
+          {|
+          {verificationKey: "%s", hash: %s}
+        |}
+          (Pickles.Side_loaded.Verification_key.to_base58_check
+             vk_with_hash.data)
+          ( Pickles.Backend.Tick.Field.to_yojson vk_with_hash.hash
+          |> Yojson.Safe.to_string )
+  in
+  let permissions (p : Mina_base.Permissions.t Snapp_basic.Set_or_keep.t) =
+    match p with
+    | Keep ->
+        "null"
+    | Set
+        { stake
+        ; edit_state
+        ; send
+        ; receive
+        ; set_delegate
+        ; set_permissions
+        ; set_verification_key
+        ; set_snapp_uri
+        ; edit_sequence_state
+        ; set_token_symbol
+        ; increment_nonce
+        } ->
+        let auth = function
+          | Mina_base.Permissions.Auth_required.None ->
+              "None"
+          | Either ->
+              "Either"
+          | Proof ->
+              "Proof"
+          | Signature ->
+              "Signature"
+          | Both ->
+              "Both"
+          | Impossible ->
+              "Impossible"
+        in
+        sprintf
+          {|
+      { stake: %s
+  , editState: %s
+  , send: %s
+  , receive: %s
+  , setDelegate: %s
+  , setPermissions: %s
+  , setVerificationKey: %s
+  , setSnappUri: %s
+  , editSequenceState: %s
+  , setTokenSymbol: %s
+  , incrementNonce: %s
+      }
+    |}
+          (if stake then "true" else "false")
+          (auth edit_state) (auth send) (auth receive) (auth set_delegate)
+          (auth set_permissions)
+          (auth set_verification_key)
+          (auth set_snapp_uri) (auth edit_sequence_state)
+          (auth set_token_symbol) (auth increment_nonce)
+  in
+  let protocol_state =
+    sprintf
+      {|
+      {
+      nextEpochData: {
+        epochLength: null, 
+        lockCheckpoint: null, 
+        startCheckpoint: null, 
+        seed: null, 
+        ledger: {
+          totalCurrency: null, 
+          hash: null}}, 
+      stakingEpochData: {
+        epochLength: null, 
+        lockCheckpoint: null, 
+        startCheckpoint: null, 
+        seed: null, 
+        ledger: {
+          totalCurrency: null, 
+          hash: null}}, 
+      globalSlotSinceGenesis: null, 
+      globalSlotSinceHardFork: null, 
+      totalCurrency: null, 
+      minWindowDensity: null, 
+      blockchainLength: null, 
+      timestamp: null, 
+      snarkedNextAvailableToken: null, 
+      snarkedLedgerHash: null}
+    |}
+  in
+  let party (p : Party.t) =
+    let authorization = authorization p.authorization in
+    let predicate =
+      match p.data.predicate with
+      | Nonce n ->
+          sprintf "{ nonce: \"%s\" }" (Account.Nonce.to_string n)
+      | Full _a ->
+          (*TODO*)
```

Comment:
Is this TODO tracked somewhere? If not, it should be.

Reply:
Issue #10008, added to the comment here.